Objects/object.c

```diff
@@ -2107,6 +2107,12 @@ _PyTrash_thread_destroy_chain(void)
          * up distorting allocation statistics.
          */
         assert(op->ob_refcnt == 0);
+        /* The deallocator executing this function still occupies
```

Comment:
> So counting it into trash_delete_nesting is natural.

That's not really what matters here, though. What the increment avoids is the possibility of an arbitrary deep recursion in `_PyTrash_thread_destroy_chain`.

> And I also want to list the example you give here. Is it okay?

Yes!